components/rule-mgt/org.wso2.carbon.identity.rule.management/src/main/java/org/wso2/carbon/identity/rule/management/service/impl/RuleManagementServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.rule.management.service.impl;
+
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.rule.management.dao.RuleManagementDAO;
+import org.wso2.carbon.identity.rule.management.dao.impl.CacheBackedRuleManagementDAO;
+import org.wso2.carbon.identity.rule.management.dao.impl.RuleManagementDAOImpl;
+import org.wso2.carbon.identity.rule.management.exception.RuleManagementClientException;
+import org.wso2.carbon.identity.rule.management.exception.RuleManagementException;
+import org.wso2.carbon.identity.rule.management.model.Rule;
+import org.wso2.carbon.identity.rule.management.service.RuleManagementService;
+
+/**
+ * Implementation of Rule Management Service.
+ */
+public class RuleManagementServiceImpl implements RuleManagementService {
+
+    private static final RuleManagementServiceImpl ruleManagementService = new RuleManagementServiceImpl();
+    private final RuleManagementDAO ruleManagementDAO;
+
+    private RuleManagementServiceImpl() {
+
+        ruleManagementDAO = new CacheBackedRuleManagementDAO(new RuleManagementDAOImpl());
+    }
+
+    public static RuleManagementServiceImpl getInstance() {
+
+        return ruleManagementService;
+    }
+
+    /**
+     * Add a new rule.
+     *
+     * @param rule         Rule to be added.
+     * @param tenantDomain Tenant domain.
+     * @return Added rule.
+     * @throws RuleManagementException If an error occurs while adding the rule.
+     */
+    @Override
+    public Rule addRule(Rule rule, String tenantDomain) throws RuleManagementException {
+
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        ruleManagementDAO.addRule(rule, tenantId);
+        return ruleManagementDAO.getRuleByRuleId(rule.getId(), tenantId);
```

Comment:
```suggestion
        return getRuleByRuleId(rule.getId(), tenantId);
```
Shall we replace ruleManagementDAO.getRuleByRuleId() with the public method ?

Reply:
It's not necessary. I rather like to access the dao layer. service layer api may have additional logics which is not quite related